### ISSUE TITLE
Some servers throw errors with HTTP/2 requests

### ISSUE THREAD

**Issue by 0xAwali** [Sh1Yo/x8#6]
### So Request Of Burp Suite Was

```
GET / HTTP/2
Host: www.google.com
Cookie: ******
```
### So When Use Burp Suite Extension "Send To" , x8 Will Parse The Request Like That

```
GET https://www.google.com HTTP/2
Host: www.google.com
Cookie: ******
```

### So If You Tried Send This Request By Using Burp , You Will Get This Response

```
HTTP/2 400 Bad Request
Content-Type: text/html; charset=UTF-8
Referrer-Policy: no-referrer
Content-Length: 1555
```

### So Every Request Sent By x8 Not Correct So x8 Should Send Request Like This

```
GET / HTTP/2
Host: www.google.com
Cookie: ******
```
### To https://www.google.com Not

```
GET https://www.google.com HTTP/2
Host: www.google.com
Cookie: ******
```
### To  https://www.google.com

**Comment by Sh1Yo** [comment]
Can you provide the version of x8, exact command, and request? Mine works correctly
```http
GET / HTTP/2
Host: www.google.com
Cookie: ******
```
Got parsed as
```http
GET / HTTP/2
Cookie: ******
Accept-Charset: utf-8, iso-8859-1;q=0.5, C0VTH;q=0.2, *;q=0.1
Host: www.google.com
Accept: */*, text/dLewk
Accept-Language: en-US, Eb6bK;q=0.9, *;q=0.5
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.97 Safari/537.36
```

**Comment by 0xAwali** [comment]
### Here Is Request 

![request](https://user-images.githubusercontent.com/54332136/125158495-e38aa380-e171-11eb-86a8-613618146def.jpg)

### Command

![command](https://user-images.githubusercontent.com/54332136/125158510-f7cea080-e171-11eb-9e63-8885b35181bc.jpg)

### xterm Output So Notice Code 400 and Result Of x8 It Will Always No Found

![output](https://user-images.githubusercontent.com/54332136/125158541-2ea4b680-e172-11eb-9952-58ac637aa2fc.jpg)

## Version

![version](https://user-images.githubusercontent.com/54332136/125158549-36fcf180-e172-11eb-9d22-ffa22783f72d.jpg)

### + It Can Not Save Output When Using -o out.txt

**Comment by Sh1Yo** [comment]
The tool doesn't print the actual request on start. To see the actual request, you can force the tool to make requests via proxy (-x flag) or use the --test flag. Add `-x http://localhost:8080 --test` to the command and show the request/response in the burp proxy tab.

**Comment by 0xAwali** [comment]
I Know , It Will Work Correctly If I Used -x Option But Just Try To Run Your Tool One Without -x And Another With -x , The Result Is Different e.g. With -x There Are Parameters Found But Without -x There Are Not Parameters Found  , Plus x8 Dose Not Save Output If I Used -o Option

**Comment by Sh1Yo** [comment]
Then add the `--test` flag and give me a screenshot.. Also, the `-o` option is not supposed to save the whole output. It saves only found parameters.

**Comment by Sh1Yo** [comment]
Also, what version of burp suite do you use? Because mine throws 400 on the provided request just as the tool does.

**Comment by 0xAwali** [comment]
### Burp Suite Version 2.7
### Raw Request
```
GET / HTTP/2
Host: www.google.com
Cookie: ******
```
### So Run With -x http://127.0.0.1:8080
### ./x8 -r raw.txt -c 3 -w parameters.txt --max 20 -x http://127.0.0.1:8080 -o output.txt

![Screenshot_2021-07-10_16_29_14](https://user-images.githubusercontent.com/54332136/125166598-aab3f400-e19c-11eb-8e46-010f53b83073.jpg)

### So Run Without -x http://127.0.0.1:8080
### ./x8 -r raw.txt -c 3 -w parameters.txt --max 20 -o output.txt

![Screenshot_2021-07-10_16_30_52](https://user-images.githubusercontent.com/54332136/125166628-d33bee00-e19c-11eb-97d2-0fb35276b7fe.jpg)

**Comment by Sh1Yo** [comment]
I can't help you without knowing all the details. Can you run the tool with the `--test` flag so I can see the exact request and response?

**Comment by Sh1Yo** [comment]
Looks like something is wrong with http/2 and the reqwest library. I will check it. For now you can use http/1.1

**Comment by 0xAwali** [comment]
Just There Is Confuse From My Side Here 

![Screenshot_2021-07-10_16_29_14(1)](https://user-images.githubusercontent.com/54332136/125167349-527ef100-e1a0-11eb-9655-8a8aef617999.jpg)

Reflected Here Is The Parameters That x8 Discovered OR Just Words Reflected Into The Response ?

**Comment by Sh1Yo** [comment]
These are real parameters, the values of which were reflected a different number of times, in contrast to non-existent ones.

**Comment by Sh1Yo** [comment]
Fixed in 2.3.2.
